### ISSUE TITLE
Unicode support for non-latin language

### ISSUE THREAD

**Issue by Sivan22** [modelcontextprotocol/python-sdk#107]
**Describe the bug**
when sending or recieving a non-latin text, the text becomes Jibrish (Non-sense or ?????)

**To Reproduce**
send a request to one of the tools with non-latin characters

**Expected behavior**
it should handle all unicode-supported languages

**Desktop (please complete the following information):**
 - OS: windows 11 home

**Comment by dsp-ant** [comment]
Do you have a repro? I am trying this and for me it works. What is your System encoding? Maybe it's releated to https://github.com/modelcontextprotocol/python-sdk/issues/123

![image](https://github.com/user-attachments/assets/ef265290-ca11-47f9-a634-5183dabce099)

**Comment by Richard-Weiss** [comment]
Have a similar issue in my Python MCP server for other unicode stuff, I added this line at the top of my server.py which fixes it:  
```
stdout.reconfigure(encoding='utf-8')
```

With mitigation suggested in #112:
![image](https://github.com/user-attachments/assets/f2132200-df8b-420f-a914-fa68fc6942ca)

With my mitigation:
![image](https://github.com/user-attachments/assets/f278d1d0-0733-4704-bde6-70ece2cd5022)

There's a different issue when it's unicode only, the message doesn't even hit my own server before erroring out from the SDK I believe:
![image](https://github.com/user-attachments/assets/f50d94a9-f25c-43cd-a7e7-2714075a6404)

Code is just this inside the tool:
```
path = arguments.path
content = arguments.content

await PathValidator.validate_file_path(path)
file_path = await PathValidator.resolve_absolute_path(path)

logger.debug(f"Writing content to {file_path}")
logger.debug(f"Content: {content}")
async with aiofiles.open(file_path, 'w') as f:
    await f.write(content)
logger.debug(f"Wrote content to {file_path}")
```

The mitigation in #112 fixes that particular issue, but I get the other encoding problem back, even with my own mitigation.
What did work for me is including `encoding='utf-8'` for file operations when using both mitigations, the debug log in the inspector won't decode it correctly though, but at least it works:
![image](https://github.com/user-attachments/assets/324c6b04-f6b0-437e-b95c-08ff4415962a)


I'm on Windows 10 and SDK version 1.2.0

**Comment by Sivan22** [comment]
thank you for that solution.
i've found another workaround using the environment variable `PYTHONIOENCODING= "utf-8" `

or using it in the mcp :
```
{
  "mcpServers": {        
      "mcp_server": {
          "command": "uv",
          "args": [
              "--directory",
              "your/path/to/directory",
              "run",
              "mcp_server"
          ],
          "env": {
            "PYTHONIOENCODING": "utf-8" 
          }
      }
  }
}
```

this solution comes from https://github.com/modelcontextprotocol/servers/issues/57

**Comment by Richard-Weiss** [comment]
Yeah, that's smart too. Claude suggested that, but I was unsure of where I could put the env variable because in the code would be too late for it to work.  
Still need to include `encoding='utf-8'` when reading a file though.

**Comment by Kludex** [comment]
Is this still an issue?

**Comment by Richard-Weiss** [comment]
I can check in 9.5 hours or so. Is the "low-level" server implementation still a thing? I imagine the sdk has changed quite a lot in the past 7 months.

**Comment by Kludex** [comment]
It's still a thing, yep.

**Comment by Richard-Weiss** [comment]
Alright, I've updated that stone old server to use mcp 1.12.0 which just worked which was really nice, then I removed my own encoding handling, works well out of the box now 👍:
<img width="3776" height="1162" alt="Image" src="https://github.com/user-attachments/assets/37fee63d-5011-4852-95e4-c7b41c8aa41f" />

<img width="1562" height="1849" alt="Image" src="https://github.com/user-attachments/assets/14b7d796-a719-4b80-9b6b-75c62db09347" />
